thunder/core/transform_common.py

```diff
@@ -854,14 +853,33 @@ def computation(x):
         ):
             new_functional_bsym.subsymbols = new_functional_bsym.subsymbols[0].subsymbols
 
-    required_copy_bsyms, swap_map_for_return = collect_required_copy_bsyms_for_args(
-        intermediate_trace,
-        removed_copy_bsyms,
-    )
     functionalized_computation_trace = from_trace(computation_trace)
-    return_bsym = new_bsyms[-1].from_bsym_swap_proxies(swap_map_for_return)
-    functionalized_computation_trace.bound_symbols = new_bsyms[:-1] + required_copy_bsyms + [return_bsym]
     functionalized_computation_trace.set_provenance(TraceProvenance("Functionalize in-place ops"))
+    required_copy_bsyms = collect_required_copy_bsyms_for_args(intermediate_trace, removed_copy_bsyms)
+
+    producer_map = producers(new_bsyms[:-1])
+    bsym_to_copy_bsyms: dict[BoundSymbol, list[BoundSymbol]] = {}
```

Comment:
I know I'm bad at this myself, but could add a comment to this saying (if I understand it correctly) that these are the bsyms not needing processing to follow the processed bsym that is the key?

I must admit I wonder a bit if this could be clearer if the two loops below could be just a loop over the bsyms doing the processing or copying.